gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java

```diff
@@ -612,6 +614,127 @@ public static Schema switchName(Schema schema, String newName) {
+        break;
+      case BOOLEAN:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case BYTES:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case DOUBLE:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case ENUM:
+        newNamespace = namespaceOverride.containsKey(schema.getNamespace()) ? namespaceOverride.get(schema.getNamespace())
+            : schema.getNamespace();
+        newSchema =
+            Schema.createEnum(schema.getName(), schema.getDoc(), newNamespace, schema.getEnumSymbols());
+        break;
+      case FIXED:
+        newNamespace = namespaceOverride.containsKey(schema.getNamespace()) ? namespaceOverride.get(schema.getNamespace())
+            : schema.getNamespace();
+        newSchema =
+            Schema.createFixed(schema.getName(), schema.getDoc(), newNamespace, schema.getFixedSize());
+        break;
+      case FLOAT:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case INT:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case LONG:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case MAP:
+        newSchema = Schema.createMap(switchNamespace(schema.getValueType(), namespaceOverride));
+        break;
+      case NULL:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case RECORD:
+        newNamespace = namespaceOverride.containsKey(schema.getNamespace()) ? namespaceOverride.get(schema.getNamespace())
+            : schema.getNamespace();
+        List<Schema.Field> newFields = new ArrayList<>();
+        if (schema.getFields().size() > 0) {
+          for (Schema.Field oldField : schema.getFields()) {
+            Field newField = new Field(oldField.name(), switchNamespace(oldField.schema(), namespaceOverride), oldField.doc(),
+                oldField.defaultValue(), oldField.order());
+            newFields.add(newField);
+          }
+        }
+        newSchema = Schema.createRecord(schema.getName(), schema.getDoc(), newNamespace,
+            schema.isError());
+        newSchema.setFields(newFields);
+        break;
+      case STRING:
+        newSchema = Schema.create(schema.getType());
+        break;
+      case UNION:
+        List<Schema> newUnionMembers = new ArrayList<>();
+        if (null != schema.getTypes() && schema.getTypes().size() > 0) {
+          for (Schema oldUnionMember : schema.getTypes()) {
+            newUnionMembers.add(switchNamespace(oldUnionMember, namespaceOverride));
+          }
+        }
+        newSchema = Schema.createUnion(newUnionMembers);
+        break;
+      default:
+        String exceptionMessage = String.format("Schema namespace replacement failed for \"%s\" ", schema);
+        LOG.error(exceptionMessage);
+
+        throw new AvroRuntimeException(exceptionMessage);
+    }
+
+    // Copy schema metadata
+    copyProperties(schema, newSchema);
```

Comment:
Seems like a miss in case of switchName(), but I would prefer to update it with another PR so that we dont break any behavior that depends on that bug :)